tests/suite/cli_misc.rs

```diff
@@ -702,6 +702,36 @@ fn toolchains_symlink() {
     });
 }
 
+// issue #3344
+/// `~/.rustup/tmp` and `~/.rustup/downloads` are permitted to be symlinks.
+#[test]
+#[cfg(any(unix, windows))]
```

Comment:
isn't this equivalent to all platforms?

Reply:
There are indeed other platforms such as wasm, although I don't know if Rustup is meaningful there or not. Similar to other tests above/below it, this one relies on the fact that we compile for Unix or Windows. I just copied the code over as I mentioned above.

Reply:
I think this is a side effect of code evolution, it isn't the pattern in general, and rustup does depend on hardlinks to operate at all. Are there platforms that have hardlinks, but not symlinks?

Reply:
@rbtcollins Looks like symlinks are also available on WASM/WASI: https://doc.rust-lang.org/std/os/wasi/fs/fn.symlink.html, so I'm also okay with removing this line. Anyway Rustup in its current form will fail to build on WASM and not all dependencies of this test work on WASM anyway.